### PR TITLE
Modify rewrite rules for apidocs subdomain

### PR DESCRIPTION
**Pull request by lzap**
The idea is to rewrite /api/latest and also /api to the new subdomain.

Do not merge yet, this hasn't been published yet.

### CODE REVIEW

File: puppet/modules/web/manifests/vhost/web.pp [hunk 1]
```diff
@@ -26,12 +26,13 @@
     { 'rewrite_rule' => ['^/wiki(.*)  https://projects.theforeman.org/wiki$1  [R,L]'] },
     { 'rewrite_rule' => ['^/events/all.ics https://community.theforeman.org/c/events/l/calendar.ics [R,L]'] },
     { 'rewrite_rule' => ['^/events/? https://community.theforeman.org/c/events/l/calendar [R,L]'] },
+    { 'rewrite_rule' => ["^/api/latest(.*) https://apidocs.theforeman.org/foreman/${stable}\$1 [R,L]"] },
+    { 'rewrite_rule' => ["^/api/(.*) https://apidocs.theforeman.org/foreman/\$1 [R,L]"] },
```
Comment (ekohl) [inline puppet/modules/web/manifests/vhost/web.pp:30]: This won't capture `https://theforeman.org/api` and if we remove the directory, I think there won't be a redirect to the trailing slash leading to a 404 for users.

Comment (lzap) [reply puppet/modules/web/manifests/vhost/web.pp:30]: But this is all I can do, I cannot want to redirect everything starting with "api". We actually do have an old page named `api.html`: https://theforeman.org/api.html and that would mask it.

**Comment by lzap** [comment]
The new site is online, the following links do work:

* https://apidocs.theforeman.org/foreman/2.4/index.html
* https://apidocs.theforeman.org/foreman/2.4/
* https://apidocs.theforeman.org/foreman/2.4
* https://apidocs.theforeman.org/foreman/latest/index.html
* https://apidocs.theforeman.org/foreman/latest/
* https://apidocs.theforeman.org/foreman/latest

So we want to redirect this:

    https://theforeman.org/api/2.4/index.html -> https://apidocs.theforeman.org/foreman/2.4/index.html

Basically all original URLs starting with `/api/` should map cleanly.

File: puppet/modules/web/manifests/vhost/web.pp (hunk 1 above)
Comment (ekohl) [reply puppet/modules/web/manifests/vhost/web.pp:30]: One easy solution:
```suggestion
    { 'rewrite_rule' => ["^/api/(.*) https://apidocs.theforeman.org/foreman/\$1 [R,L]"] },
    { 'rewrite_rule' => ["^/api$ https://apidocs.theforeman.org/foreman/ [R,L]"] },
```

You'll also see that there's a single regex line as well but I'll leave that as an excise to the reader :)

Comment (lzap) [reply puppet/modules/web/manifests/vhost/web.pp:30]: Sure I can do that, amended.

**Comment by lzap** [comment]
`WARNING: double quoted string containing no variables on line 31`

You gotta be kidding me. You suffer through this voluntarily? Fixed :-)

**Comment by lzap** [comment]
Tests are passing finally.

**Comment by lzap** [comment]
Just checking - any other concerns? We are working on finalizing the theforeman.org patch so I want to know if this would be RTM. Thanks!

**Review by tbrisker — APPROVED** [review]
lgtm, theforeman.org part was merged.

**Review by ekohl — APPROVED** [review]

File: puppet/modules/web/manifests/vhost/web.pp [hunk 2]
```diff
@@ -26,12 +26,14 @@
     { 'rewrite_rule' => ['^/wiki(.*)  https://projects.theforeman.org/wiki$1  [R,L]'] },
     { 'rewrite_rule' => ['^/events/all.ics https://community.theforeman.org/c/events/l/calendar.ics [R,L]'] },
     { 'rewrite_rule' => ['^/events/? https://community.theforeman.org/c/events/l/calendar [R,L]'] },
+    { 'rewrite_rule' => ["^/api/latest(.*) https://apidocs.theforeman.org/foreman/${stable}\$1 [R,L]"] },
+    { 'rewrite_rule' => ['^/api/(.*) https://apidocs.theforeman.org/foreman/\$1 [R,L]'] },
```
Comment (ekohl) [inline puppet/modules/web/manifests/vhost/web.pp:30]: Doesn't look like this works:
```console
$ curl https://theforeman.org/api/2.4/apidoc/v2/compute_profiles/create.html -si | grep Location
Location: https://apidocs.theforeman.org/foreman/$1
```

Should have been just `$1`.

Comment (ekohl) [reply puppet/modules/web/manifests/vhost/web.pp:30]: https://github.com/theforeman/foreman-infra/pull/1608

Comment (lzap) [reply puppet/modules/web/manifests/vhost/web.pp:30]: Damn sorry!
